tests/pandas_utils/test_wide_df_handling.py

```diff
@@ -0,0 +1,388 @@
+# -*- coding: utf-8 -*-
+"""Tests for the {} module"""
+# Built-Ins
+import dataclasses
+import operator
+from typing import Any, Callable
+
+# Third Party
+import numpy as np
+import pandas as pd
+import pytest
+
+# Local Imports
+from caf.toolkit import pandas_utils as pd_utils
+
+# # # CONSTANTS # # #
+
+
+# # # CLASSES # # #
+@dataclasses.dataclass
+class WideMatrixData:
+    """Store testing data for a wide matrix.
+
+    Some helper functions to replicate the functions we're testing at their
+    most basic. This should make the tests more readable.
+    """
+
+    matrix: pd.DataFrame
+
+    def index_mask(self, selection: list[int]) -> np.ndarray:
+        """Get the mask for index matching in the shape of self.matrix."""
+        mask = np.isin(self.matrix.index.to_numpy(), selection)
+        return np.broadcast_to(mask, self.matrix.shape).T
+
+    def col_mask(self, selection: list[int]) -> np.ndarray:
+        """Get the mask for col matching in the shape of self.matrix."""
+        mask = np.isin(self.matrix.columns.to_numpy(), selection)
+        return np.broadcast_to(mask, self.matrix.shape)
+
+    def full_mask(
+        self,
+        index_select: list[int],
+        col_select: list[int],
+        join_fn: Callable,
+    ) -> np.ndarray:
+        """Get the full mask of rows and columns in the shape of self.matrix."""
+        return join_fn(self.col_mask(col_select), self.index_mask(index_select))
+
+
+@dataclasses.dataclass
+class ReportMatrixData(WideMatrixData):
+    """Store testing data for the wide_matrix_internal_external_report()"""
+
+    expected_report: pd.DataFrame
+    int_select: list[int]
+    ext_select: list[int]
+
+    def get_kwargs(self) -> dict[str, Any]:
+        """Produce a kwarg dict. Prevents repeated code."""
+        return {
+            "df": self.matrix,
+            "int_select": self.int_select,
+            "ext_select": self.ext_select,
+        }
+
+
+# # # FIXTURES # # #
+def random_matrix(shape: tuple[int,]) -> pd.DataFrame:
+    """Produce a random matrix of a given shape"""
+    rng = np.random.default_rng()
+    return pd.DataFrame(rng.random(shape))
+
+
+@pytest.fixture(name="random_square_matrix", scope="function")
+def fixture_random_square_matrix(request) -> WideMatrixData:
+    """Fixture to create random matrix of set dimensions"""
+    shape = (request.param, request.param)
+    return WideMatrixData(matrix=random_matrix(shape))
+
+
+@pytest.fixture(name="non_square_matrix", scope="module")
+def fixture_non_square_matrix(request) -> WideMatrixData:
+    """Fixture to create random matrix of set dimensions"""
+    return WideMatrixData(matrix=random_matrix(request.param))
+
+
+# # # TESTS # # #
+class TestGetWideMask:
+    """Tests for get_wide_mask()"""
+
+    # TODO(BT): Add tests for datatypes. Different + fuzzy matching. Hoping some
```

Comment:
All these tests seem to be calculating the expected matrix, I think it might be worth adding one or two hardcoded matrix results for testing just to make sure there aren't any potential issues in the numpy code which could hide problems in our tests.

Reply:
Technically this is covered in the doctests. Are you happy accepting that they cover that purpose, or would you rather they were repeated here?

Reply:
Completely forgot about those, I don't think there's need to repeat them